File: YFDebugPannel/YFDebugPannel/HCPresentationRequest.m
#import "HCPresentationRequest.h"

@implementation HCPresentationAction

+ (instancetype)actionWithTitle:(NSString *)title value:(nullable NSString *)value {
    HCPresentationAction *action = [[HCPresentationAction alloc] init];
    action.title = title;
    action.value = value;
    return action;
}

@end

@implementation HCPresentationRequest

+ (instancetype)toastWithMessage:(NSString *)message {
    HCPresentationRequest *request = [[HCPresentationRequest alloc] init];
    request.type = HCPresentationTypeToast;
    request.title = message;
    request.actions = @[];
    return request;
}

+ (instancetype)alertWithTitle:(NSString *)title message:(NSString *)message actions:(NSArray<HCPresentationAction *> *)actions {
    HCPresentationRequest *request = [[HCPresentationRequest alloc] init];
    request.type = HCPresentationTypeAlert;
    request.title = title;
    request.message = message;
    request.actions = actions ?: @[];
    return request;
}

+ (instancetype)actionSheetWithTitle:(NSString *)title message:(nullable NSString *)message actions:(NSArray<HCPresentationAction *> *)actions {
    HCPresentationRequest *request = [[HCPresentationRequest alloc] init];
    request.type = HCPresentationTypeActionSheet;
    request.title = title;
    request.message = message;
    request.actions = actions ?: @[];
    return request;
}

@end

File: YFDebugPannel/YFDebugPannel/HCPresentationRequest.swift
import Foundation

enum HCPresentationType: Int {
    case alert
    case actionSheet
    case toast
}

final class HCPresentationAction {
    var title: String
    var value: String?

    init(title: String, value: String?) {
        self.title = title
        self.value = value
    }

    static func action(title: String, value: String?) -> HCPresentationAction {
        HCPresentationAction(title: title, value: value)
    }
}

final class HCPresentationRequest {
    var type: HCPresentationType = .toast
    var title: String = ""
    var message: String?
    var actions: [HCPresentationAction] = []

    static func toast(message: String) -> HCPresentationRequest {
        let request = HCPresentationRequest()
        request.type = .toast
        request.title = message
        request.actions = []
        return request
    }

    static func alert(title: String, message: String, actions: [HCPresentationAction]) -> HCPresentationRequest {
        let request = HCPresentationRequest()
        request.type = .alert
        request.title = title
        request.message = message
        request.actions = actions
        return request
    }

    static func actionSheet(title: String, message: String?, actions: [HCPresentationAction]) -> HCPresentationRequest {
        let request = HCPresentationRequest()
        request.type = .actionSheet
        request.title = title
        request.message = message
        request.actions = actions
        return request
    }
}
